tests/assoc_sim.R

```diff
@@ -0,0 +1,140 @@
+  }
+  
+  # Pr[var2 = N + 1 | var1 = N] = 0.5
+  # Pr[var2 = N     | var1 = N] = 0.5
+  samples[[2]] <- samples[[1]] + sample(c(0, 1), N, replace = TRUE)
+  
+  # Replace sample i with candidate i for each variable
+  for(i in 1:2) {
+    if(max(samples[[i]]) <= length(candidates[[i]])) {
+      samples[[i]] <- candidates[[i]][samples[[i]]]
+    } else {
+      # Pad candidates with sample strings
+      # If only 2 candidates, new set of candidates becomes:
+      # candidate1, candidate2, str3, str4, ...
+      len <- length(candidates)
+      candidates[[i]][(len + 1):max(samples[[i]])] <- apply(
+        as.matrix((len + 1):max(samples[[i]])),
+        1,
+        function(i) paste("str", as.character(i), sep = ""))
+      samples[[i]] <- candidates[[i]][samples[[i]]]
+    }
+  }
+  
+  # Randomly assign cohorts in each dimension
+  cohorts <- sample(1:m, N, replace = TRUE)
+  
+  # Create and write map into mapfile_1.csv and mapfile_2.csv
+  map <- lapply(1:2, function(i) CreateMap(candidates[[i]], params))
+  write.table(map[[1]]$map_pos, file = paste(mapfile, "_1.csv", sep = ""),
+              sep = ",", col.names = FALSE, na = "", quote = FALSE)
+  write.table(map[[2]]$map_pos, file = paste(mapfile, "_2.csv", sep = ""),
+              sep = ",", col.names = FALSE, na = "", quote = FALSE)
+  
+  # Write reports into reportsfile.csv
+  # Format:
+  #     cohort, bloom filter var1, bloom filter var2
+  reports <- lapply(1:2, function(i)
+    EncodeAll(samples[[i]], cohorts, map[[i]]$map, params))
+  # Organize cohorts and reports into format
+  write_matrix <- cbind(as.matrix(cohorts),
+                        as.matrix(lapply(reports[[1]], 
+                            function(x) paste(x, collapse = ""))),
+                        as.matrix(lapply(reports[[2]],
+                            function(x) paste(x, collapse = ""))))
+  write.table(write_matrix, file = reportsfile, quote = FALSE,
+              row.names = FALSE, col.names = FALSE, sep = ",")
+}
+
+# Command line arguments
+spec = matrix(c(
+  "candidates", "c", 2, "character",
+  "params", "p", 2, "character",
+  "reports", "r", 2, "character",
+  "map", "m", 2, "character",
+  "num", "n", 2, "integer",
+  "unif", "u", 2, "logical",
+  "help", "h", 0, "logical"
+  ), byrow = TRUE, ncol = 4)
+opt = getopt(spec)
+
+# Usage
+if (!is.null(opt$help)) {
+  cat(getopt(spec, usage = TRUE))
+  q(status = 1)
+}
+
+# Defaults
+if (is.null(opt$candidates))  {opt$candidates = "candidates.csv"}
+if (is.null(opt$params))      {opt$params = "params.csv"}
+if (is.null(opt$reports))     {opt$reports = "reports.csv"}
+if (is.null(opt$map))         {opt$map = "map"}
+if (is.null(opt$num))         {opt$num = 1e05}
+if (is.null(opt$unif))        {opt$unif = FALSE}
+
+ptm <- proc.time()
+
+candidates <- GetCandidatesFromFile(opt$candidates)
+params <- ReadParameterFile(opt$params)
+SimulateReports(opt$num, candidates, params, opt$map, opt$reports, opt$unif)
```

Comment:
Does this file do the analogous thing to  gen_counts.R ?

If so we should probably name them in parallel, like

gen_dist.R
gen_assoc.R

And also

analyze_dist.R
analyze_assoc.R